Plugins/FormatSwift/Plugin.swift

```diff
@@ -166,3 +167,92 @@ enum CommandError: Error {
   case lintFailure
   case unknownError(exitCode: Int32)
 }
+
+// MARK: - SwiftVersion
+
+struct SwiftVersion: Comparable {
+  var major: Int
+  var minor: Int
+
+  static func ==(_ lhs: SwiftVersion, _ rhs: SwiftVersion) -> Bool {
+    lhs.major == rhs.major
+      && lhs.minor == rhs.minor
+  }
+
+  static func <(_ lhs: SwiftVersion, _ rhs: SwiftVersion) -> Bool {
+    if lhs.major == rhs.major {
+      return lhs.minor < rhs.minor
+    } else {
+      return lhs.major < rhs.major
+    }
+  }
+}
+
+extension Package {
+
+  // MARK: Internal
+
+  /// The minimum Swift version supported by this package
+  var minimumSwiftVersion: SwiftVersion {
+    supportedSwiftVersions.sorted().first!
+  }
+
+  // MARK: Private
+
+  /// Swift versions supported by this package. Guaranteed to be non-empty.
+  ///  - This includes the `swift-tools-version` from the `Package.swift`,
+  ///    plus the Swift version of any additional version-specific Package manifest
+  ///    (e.g. `Package@swift-5.6.swift`).
+  private var supportedSwiftVersions: [SwiftVersion] {
+    guard let projectDirectory = URL(string: directory.string) else { return [] }
```

Comment:
@calda can you help me follow where the `directory` property is defined? I didn't see it listed here: https://developer.apple.com/documentation/packagedescription/package.

Reply:
I think this is the `PackagePlugin.Package` type, which is different from `PackageDescription.Package`. I can't actually find documentation online for the `PackagePlugin` framework, but it exposes the APIs that you use in SPM package plugins.